### PR TITLE
fix: align relay payment polling contract with tx-schemas

### PR DESCRIPTION
**Pull request by whoabuddy**
## Summary
- align agent-news payment parsing and polling responses with `@aibtc/tx-schemas@^0.3.0`
- keep caller-facing pending states canonical (`queued`, `broadcasting`, `mempool`) and preserve normalized `terminalReason`
- add confirmed-only staging/reconciliation for brief access and classifieds, keyed by relay-owned `paymentId`
- surface `checkStatusUrl` on pending and polling responses when the relay provides it
- staged payment records expire after 24h if they never reach terminal state

## Contract alignment completed
- replaced local RPC/HTTP payment result typings with shared `tx-schemas` types
- validate relay RPC submit/check payloads with shared schemas before use
- centralize `/api/payment-status/:paymentId` response shaping through the shared HTTP polling schema
- keep `submitted` out of caller-facing responses and docs
- keep `terminalReason` as the normalized terminal signal
- keep confirmed-only delivery as the default by staging pending work locally and finalizing only on `confirmed`

## Simplifications
- removed duplicated ad-hoc payment-status response shaping in the route
- removed now-redundant local relay result interfaces covered by `tx-schemas`
- tightened tests around canonical status handling, `terminalReason`, `checkStatusUrl`, and staging behavior
- dropped the stray local package tarball from the branch before PR

## Compatibility / rollout notes
- brief pending responses still emit `X-Payment-Status` and `X-Payment-Id` as a measured compatibility shim; runtime/docs now describe canonical polling as `checkStatusUrl` when present, otherwise `/api/payment-status/:paymentId`
- this repo assumes the updated `@aibtc/tx-schemas` release and depends on the relay surfacing canonical caller-facing states plus optional `checkStatusUrl`

## Verification
- `npm run typecheck` — clean
- `npm run lint` — pre-existing warnings only, no errors
- `npm test` — 27 test files, 246 tests, all passing (intermittent workerd timeouts on full suite are the known `Failed to get handler to worker` issue, not related to this PR)
- payment-specific tests: `payment-staging`, `pending-payment-route-guards`, `payment-status-response`, `payment-logging`, `x402-rpc`, `x402-error-mapping`, `x402-http-fallback`, `payment-status-route` — 8 files, 45 tests, all passing

## Residual risk / follow-up
- the legacy brief pending headers remain until downstream callers are confirmed to use canonical polling fields

### CODE REVIEW

File: src/routes/payment-status.ts [hunk 1]
```diff
@@ -28,42 +31,73 @@ paymentStatusRouter.get("/api/payment-status/:paymentId", async (c) => {
...
-    return c.json(
-      { error: "Payment not found — it may have expired or the id is incorrect", paymentId },
-      404
-    );
+  if (body.status === "confirmed" || body.status === "failed" || body.status === "replaced" || body.status === "not_found") {
+    const reconcileResult = await reconcilePaymentStage(c.env, paymentId, {
+      status: body.status,
+      txid: body.txid,
+      terminalReason: body.terminalReason,
+    });
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply src/routes/payment-status.ts]: Fixed in `56ec4e9`. Terminal polling responses now return `503` with a reconciliation-pending warning when `reconcilePaymentStage()` throws or returns a non-ok result, and the route tests cover both the happy path and the retry case.

File: src/routes/classifieds.ts [hunk 2]
```diff
@@ -210,40 +230,92 @@ classifiedsRouter.post(
...
+        classified_id: provisionalClassifiedId,
+        btc_address,
+        category,
+        headline: sanitizeString(headline, 100),
+        body: adBody ? sanitizeString(adBody, 500) : null,
+        payment_txid: verification.txid ?? null,
+      },
+    });
+    if (!stageResult.ok || !stageResult.data) {
+      return c.json({ error: stageResult.error ?? "Failed to stage classified submission" }, stageResult.status ?? 500);
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply src/routes/classifieds.ts]: Not taking the async/deferred staging behavior in this PR for the same reason as brief access: confirmed payment should not be reported as delivered until the durable write succeeds. The route still fails closed on staging errors today; if we relax that, it should be a deliberate availability tradeoff in a follow-up change.

File: src/objects/news-do.ts [hunk 3]
```diff
@@ -390,6 +430,138 @@ export class NewsDO extends DurableObject<Env> {
...
+    // -------------------------------------------------------------------------
+
+    this.router.post("/payment-staging", async (c) => {
+      const body = await parseRequiredJson<{ paymentId?: string; payload?: PaymentStagePayload }>(c);
+      if (!body?.paymentId || !body.payload?.kind) {
+        return c.json(
+          { ok: false, error: "Missing required fields: paymentId, payload.kind" } satisfies DOResult<PaymentStageMaterialized>,
+          400
+        );
+      }
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply src/objects/news-do.ts]: Fixed in `d81bf9b`. `/payment-staging` now rejects unsupported `payload.kind` values up front with `400`, and there’s a new test covering that boundary.

File: src/objects/schema.ts [hunk 4]
```diff
@@ -110,6 +110,19 @@ CREATE TABLE IF NOT EXISTS referral_credits (
   created_at       TEXT DEFAULT (datetime('now'))
 );
 
+CREATE TABLE IF NOT EXISTS payment_staging (
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply src/objects/schema.ts]: Agreed this is worth doing, but I’m leaving TTL cleanup out of this PR. It’s orthogonal to the contract-alignment fixes and needs a separate decision on retention window plus when the sweep should run in the DO lifecycle.

**Review by tfireubs-ui — APPROVED** [review]
LGTM — solid refactor. Payment staging + confirmed-only delivery semantics correctly prevent premature finalization. Zod schema validation at the relay boundary is the right approach. Test coverage is comprehensive (staging lifecycle, reconciliation edge cases, 503 retry).

Minor: buildPaymentStatusResponse logs validation errors via console.error, which is sufficient for debugging.

Approved.
